.github/workflows/non_docker_build.yml

```
@@ -154,7 +154,16 @@ jobs:
 
   build_wheels:
     name: Build wheels
-    runs-on: ubuntu-24.04
+    # Trial running on RunsOn self-hosted runners (https://runs-on.com). The label
+    # list selects an 8-vCPU c7i EC2 instance with the ubuntu24-full image; the
+    # extras=s3-cache label enables the RunsOn S3-backed Magic Cache, which the
+    # runs-on/action step below wires into actions/cache.
+    runs-on:
+    - runs-on=${{ github.run_id }}
+    - cpu=8
```

actionlint (1.7.12)

[error] label "cpu=8" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2025-vs2026", "windows-2022", "windows-11-arm", "ubuntu-slim", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "macos-latest", "macos-latest-xlarge", "macos-latest-large", "macos-26-intel", "macos-26-xlarge", "macos-26-large", "macos-26", "macos-15-intel", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xlarge", "macos-14-large", "macos-14", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file

(runner-label)